specs/online_order_spec.rb

```diff
@@ -2,51 +2,81 @@
 require 'minitest/reporters'
 require 'minitest/skip_dsl'
 
-# TODO: uncomment the next line once you start wave 3
-# require_relative '../lib/online_order'
+require_relative '../lib/online_order'
+require_relative '../lib/order'
+require_relative '../lib/customer'
 # You may also need to require other classes here
 
 # Because an OnlineOrder is a kind of Order, and we've
 # already tested a bunch of functionality on Order,
 # we effectively get all that testing for free! Here we'll
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
   describe "#initialize" do
     it "Is a kind of Order" do
       # Check that an OnlineOrder is in fact a kind of Order
+      id = 1313
+      products = "burning sage"
+      customer = Grocery::Customer.new(1001, "bad-dog@grounded.net", "123 Cat Paradise, Seattle, WA 98101")
 
-      # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      online_order = Grocery::OnlineOrder.new(id, products, customer)
+
+      online_order.must_be_kind_of Grocery::Order
     end
 
     it "Can access Customer object" do
-      # TODO: Your test code here!
+      id = 1313
+      products = "burning sage"
+      customer = Grocery::Customer.new(1001, "bad-dog@grounded.net", "123 Cat Paradise, Seattle, WA 98101")
+
+      online_order = Grocery::OnlineOrder.new(id, products, customer)
+      online_order.customer.email.must_equal "bad-dog@grounded.net"
     end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+      id = 1313
+      products = "burning sage"
+      customer = Grocery::Customer.new(1001, "bad-dog@grounded.net", "123 Cat Paradise, Seattle, WA 98101")
+
+      online_order = Grocery::OnlineOrder.new(id, products, customer)
+      online_order.status.must_equal :pending
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      order = Grocery::OnlineOrder.new(1337, products, "Finn")
+
+      sum = products.values.inject(0, :+)
+      expected_total = sum + (sum * 0.075).round(2) + 10
+
+      order.total.must_equal expected_total
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      products = {}
+      order = Grocery::OnlineOrder.new(1337, products, "Finn")
+
+      expected_total = 0 # no products plus no shipping
+      order.total.must_equal expected_total
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+      order = Grocery::OnlineOrder.new(666, {1 => "cake",}, "Finn", :shipped)
+
+      proc{order.add_product(1, "cake")}.must_raise ArgumentError
```

Comment:
You should also check processing and completed statuses here, since that's what the test's name says.